sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -667,6 +669,13 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
 
       // For debugging dump some statistics about how much time was spent in various optimizer rules
       logWarning(RuleExecutor.dumpTimeSpent())
+
+      val codegenInfo =
+        s"""
+           |=== Metrics of Whole Codegen ===
+           |Total compile time: ${CodeGenerator.compileTime.sum.toDouble / NANOS_PER_SECOND} seconds
```

Comment:
Probably, we need to reset this metric in `beforeAll`? btw , could we reuse `CodegenMetrics.METRIC_COMPILATION_TIME` for this purpose? Getting the same metric for the same purpose seems redundant.

Reply:
At first, I wanted to use` CodegenMetrics.METRIC_COMPILATION_TIME` to do this. After many experiments, I found that `CodegenMetrics.METRIC_COMPILATION_TIME` is only effective for a single test case, and cannot play a role in the whole life cycle of `SQLQueryTestSuite`.
Do you know the reason?

Reply:
@maropu `Histogram` is a metric which calculates the distribution of a value.